drv/sidecar-front-io/src/leds.rs

```diff
@@ -319,19 +394,18 @@ impl Leds {
             };
 
             match err {
-                LedErr::OpenCircuit => summary.open_circuit |= port_mask,
-                LedErr::ShortCircuit => summary.short_circuit |= port_mask,
-                LedErr::Invalid => summary.invalid |= port_mask,
+                LedErr::OpenCircuit => summary.open_circuit |= index,
+                LedErr::ShortCircuit => summary.short_circuit |= index,
+                LedErr::Invalid => summary.invalid |= index,
                 LedErr::NoError => (),
             }
         }
 
-        // handle the system LED outside the loop since it is the 33rd index
-        let sys_led_loc = LED_MAP[SYSTEM_LED_IDX];
-        let sys_output: usize = sys_led_loc.output as usize;
-        summary.system_led_err = match sys_led_loc.controller {
-            LedController::Left => left_errs.errors[sys_output],
-            LedController::Right => right_errs.errors[sys_output],
+        summary.system_led_err = match SYSTEM_LED.controller {
+            LedController::Left => left_errs.errors[SYSTEM_LED.output as usize],
+            LedController::Right => {
+                right_errs.errors[SYSTEM_LED.output as usize]
+            }
         };
```

Comment:
you could avoid overflowing here with something like
```suggestion
        summary.system_led_err = match SYSTEM_LED.controller {
            LedController::Left => &left_errs,
            LedController::Right => &right_errs,
        }
        .errors[SYSTEM_LED.output as usize];
```

(the same trick is applicable elsewhere in the file; up to you how widely you want to use it)